### PR TITLE
Added HRRR method for 2-m diagnostics, added several diagnostics variables to the history files

### PR DESCRIPTION
**Pull request by tanyasmirnova**
## Description
The changes in this PR affect the FV3_HRRR physics suite. They include:

Flux-based method for 2-m diagnostics (sfc_diag.f). This diagnostics will be enabled when MYNN surface layer
scheme and RUC LSM are used in the physics suite.
Several variables are added to the output for FV3_HRRR physics suite. Some of the variables have to be moved from Interstitial to Diag. Some clean-up of variables description and units are made in GFS_diagnostics.F90.
The answers will be changed with this PR.

**Edit from Sam Trahan**: This PR now contains the changes from #142 which provide a fast alternative to fms2 io when reading physics restarts in FV3GFS_io.F90. The #142 changes will not change results.

## Testing

The changes has been performed on jet. The new baseline is in /mnt/lfs1/BMC/wrfruc/smirnova/RT_BASELINE/Tanya.Smirnova/FV3_RT/REGRESSION_TEST_INTEL.
The log RegressionTests_jet.intel.log is committed.

## Dependencies
https://github.com/NOAA-GSL/ccpp-physics/pull/146

### CODE REVIEW

**Comment by SamuelTrahanNOAA** [comment]
This PR will be combined with #142 which does not change results.

**Review by mdtoyNOAA — APPROVED** [review]

**Review by hannahcbarnes — APPROVED** [review]

**Comment by SamuelTrahanNOAA** [comment]
Bad news: this PR breaks restart support: rap_restart and rap_sfcdiff_restart both fail. This is true even if I revert Sam Trahan's changes.

**Review by joeolson42 — APPROVED** [review]
I don't approve the diagnostic change but I'll approve the PR anyway due to time constraints. I hope we find a more sensible solution in the near future.

**Comment by SamuelTrahanNOAA** [comment]
> I don't approve the diagnostic change but I'll approve the PR anyway due to time constraints. I hope we find a more sensible solution in the near future.

Regardless of your feelings on the diagnostic change, this PR can't go in if it breaks restart support.

**Comment by SamuelTrahanNOAA** [comment]
> I don't approve the diagnostic change but I'll approve the PR anyway due to time constraints. I hope we find a more sensible solution in the near future.

Also, if we add changes that the community deems dubious, then a future gsl/develop PR to the community may be rejected. If this PR (#140) will be problematic, we should fix it before the next merge of RRFS_dev to gsl/develop.

File: .gitmodules
Comment (SamuelTrahanNOAA) [inline .gitmodules]: Code manager will need to revert this before commit.

**Pull request merged**
